tests/io/test_GridOperations.py

```diff
@@ -0,0 +1,286 @@
+            nc_file = netCDF4.Dataset(temp_file.name, mode='w', diskless=True)
+
+            # Define dimensions
+            nc_file.createDimension('time', 0)  # 3 time dimensions will be defined later (0=unlimited)
+            nc_file.createDimension('face', 2)  # 2 faces
+
+            # Create variables
+            mesh2d = nc_file.createVariable('mesh2d', 'f4')  # 'f4' specifies the data type as float32
+            mesh2d.setncattr('cf_role', "mesh_topology")
+            mesh2d.setncattr('topology_dimension', 2)
+
+            discharge = nc_file.createVariable('discharge', 'f4', ('time', 'face'))
+            discharge.setncattr('standard_name', 'discharge')
+            discharge.setncattr('mesh', mesh2d.name)
+            discharge.setncattr('location', 'face')
+
+            discharge[:] = [[27, 76],
+                            [801, 802],
+                            [214, 588]]
+
+            with mock.patch('netCDF4.Dataset') as mock_nc_dataset:
+                mock_nc_dataset.return_value = nc_file
+                data = GridOperations.read_fm_map(filename="mock.nc", varname='discharge')
+                assert numpy.array_equal(data, [214,588])
+
+    def test_read_fm_map_instantiate_dataset_in_memory_get_indexed_double_var(self):
+        with tempfile.NamedTemporaryFile(suffix='_map.nc') as temp_file:
+            nc_file = netCDF4.Dataset(temp_file.name, mode='w', diskless=True)
+
+            # Define dimensions
+            nc_file.createDimension('time', 0)  # 3 time dimensions will be defined later (0=unlimited)
+            nc_file.createDimension('face', 2)  # 2 faces
+
+            # Create variables
+            mesh2d = nc_file.createVariable('mesh2d', 'f4')  # 'f4' specifies the data type as float32
+            mesh2d.setncattr('cf_role', "mesh_topology")
+            mesh2d.setncattr('topology_dimension', 2)
+
+            discharge = nc_file.createVariable('discharge', 'f4', ('time', 'face'))
+            discharge.setncattr('standard_name', 'discharge')
+            discharge.setncattr('mesh', mesh2d.name)
+            discharge.setncattr('location', 'face')
+
+            discharge[:] = [[27, 76],
+                            [801, 802],
+                            [214, 588]]
+
+            with mock.patch('netCDF4.Dataset') as mock_nc_dataset:
+                mock_nc_dataset.return_value = nc_file
+                data = GridOperations.read_fm_map(filename="mock.nc", varname='discharge', ifld=2)
+                assert numpy.array_equal(data, [27,76])
+
+    def test_read_fm_map_from_mocked_dataset_x_coordinates_of_faces_by_projection_x_coordinate(self):
+        """
+        Testing read_fm_map: x coordinates of the faces by projection_x_coordinate.
+        """
+        filename = "mocked_file_name.nc"
+        varname = "x"
+        with mock.patch('netCDF4.Dataset') as netCDF4Dataset:
+            netCDF4Dataset.return_value = netCDF4.Dataset
```

Comment:
can this part maybe be moved to a helper method?

Reply:
it seems it is reused quite a lot

Reply:
I agree @MRVermeulenDeltares. The refactoring of readability should be done in separate issue

Reply:
https://issuetracker.deltares.nl/browse/DFAST-111